xdsl/dialects/snitch_runtime.py

```diff
@@ -235,6 +235,231 @@ class DmaWaitAllOp(SnitchRuntimeBarrier):
     name = "snrt.dma_wait_all"
 
 
+"""
+The different SSR data movers.
+    SNRT_SSR_DM0 = 0,
+    SNRT_SSR_DM1 = 1,
+    SNRT_SSR_DM2 = 2,
+"""
+ssr_dm = i32
+
+"""
+The different dimensions.
+    SNRT_SSR_1D = 0,
+    SNRT_SSR_2D = 1,
+    SNRT_SSR_3D = 2,
+    SNRT_SSR_4D = 3,
+"""
+ssr_dim = i32
+
+
+@irdl_op_definition
+class SsrLoop1dOp(SnitchRuntimeBaseOp, ABC):
```

Comment:
@superlopuh  @PapyChacal is there any way of making this op more generic?
Such that we have one base class that constructs all of these 1D to 4D ops?
It's 4 times the same pattern, but might be more difficult to make this work for multiple cases?

Reply:
I agree with this.
How about using `var_operand_def`?
As you pointed out stencil use this for example [here](https://github.com/xdslproject/xdsl/blob/0b866aa5f94d980eb51c2fe0dc7eddf9c784b409/xdsl/dialects/stencil.py#L321).